### PR TITLE
export exif tag: use rational type when necessary

### LINKED ISSUE #10399: Problem with export "Subject Distance" exif tag
Hello.

My RAWs have a "Focus Distance" tag

```
~/temp$ exiftool 20211016__DSC5125.NEF | grep Distance
Focus Distance                  : 3.16 m
```
image_src

I am trying to use it as a value for  "Subject Distance" exported exif tag.
pref1

There is no value in the resulting jpeg
```
~/temp$ exiftool 20211016__DSC5125.jpg | grep Subject
Subject Distance                :  m
Subject Distance Range          : Unknown

~/temp$ exiftool 20211016__DSC5125.jpg | grep 'Lens Model'
Lens Model                      : Tamron SP AF 150-600mm F/5-6.3 VC USD G2
```

If I specify variable _$(EXIF_FOCUS_DISTANCE)_ as the value for "Lens Model" exif tag
pref2

then it is exported to this tag.
```
~/temp$ exiftool 20211016__DSC5125_01.jpg | grep 'Lens Model'
Lens Model                      : 3.16

~/temp$ exiftool 20211016__DSC5125_01.jpg | grep Subject
Subject Distance                :  m
Subject Distance Range          : Unknown

```

darktable version: master
OS:  Debian 11 /  Windows 7

### PR DESCRIPTION
**Pull request by phweyland**
fixes #10399

### CODE REVIEW

File: src/common/exif.cc [hunk 1]
```diff
@@ -4037,7 +4037,27 @@ int dt_exif_xmp_attach_export(const int imgid, const char *filename, void *metad
                 else iptcData[tagname] = result;
               }
               else if(g_str_has_prefix(tagname, "Exif."))
+              {
+                const char *type = _exif_get_exiv2_tag_type(tagname);
+                if(!g_strcmp0(type, "Rational"))
+                {
+                  float float_value = (float)std::atof(result);
+                  if(!std::isnan(float_value)
```
Comment (esq4) [inline src/common/exif.cc:4045]: `if(!std::isnan(float_value) )` 
________________________________^-------

Comment (phweyland) [reply src/common/exif.cc:4045]: right ...

**Comment by esq4** [comment]
This works on Windows.

**Comment by phweyland** [comment]
> This works on Windows.

Good to know. Thanks for reporting and ... testing!

File: src/common/exif.cc [hunk 2]
```diff
@@ -4037,7 +4037,27 @@ int dt_exif_xmp_attach_export(const int imgid, const char *filename, void *metad
                 else iptcData[tagname] = result;
               }
               else if(g_str_has_prefix(tagname, "Exif."))
+              {
+                const char *type = _exif_get_exiv2_tag_type(tagname);
+                if(!g_strcmp0(type, "Rational"))
```
Comment (kmilos) [inline src/common/exif.cc:4042]: Could be signed "SRational" as well...

**Comment by esq4** [comment]
> This works on Windows.

Works on Linux too.

File: src/common/exif.cc (hunk 2 above)
Comment (phweyland) [reply src/common/exif.cc:4042]: This is right.

**Comment by phweyland** [comment]
@Turbogit Pascal, this PR should be ok for your review.

**Review by TurboGit — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**